besu/src/test/java/org/hyperledger/besu/cli/BesuCommandTest.java

```diff
@@ -2338,12 +2338,33 @@ public void pruningIsEnabledWhenSpecified() throws Exception {
   }
 
```

Comment:
Do we need a test that the pruning override logic works? When using fast sync pruning should be true, unless the user overrides it, right?

Maybe this has been covered by `pruningEnabledExplicitly` and `pruningDisabledExplicitly `.

Reply:
Good point, I'll add the contrary sync modes to the explicit tests to exercise that logic.

Reply:
Added